helios-services/src/test/java/com/spotify/helios/servicescommon/RuleBasedZooKeeperAclProviderTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.servicescommon;
+
+import com.google.common.collect.Sets;
+
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.data.Id;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.apache.zookeeper.ZooDefs.Perms.DELETE;
+import static org.apache.zookeeper.ZooDefs.Perms.CREATE;
+import static org.apache.zookeeper.ZooDefs.Perms.READ;
+import static org.apache.zookeeper.ZooDefs.Perms.WRITE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class RuleBasedZooKeeperAclProviderTest {
+
+  @Test
+  public void testSimple() {
+    final Id id1 = new Id("some_scheme", "id1");
+    final Id id2 = new Id("some_scheme", "id2");
+    final RuleBasedZooKeeperAclProvider aclProvider = RuleBasedZooKeeperAclProvider.builder()
+        .rule("/foo/baz", DELETE, id1)
+        .rule("/foo/bar", CREATE, id1)
+        .rule("/foo/qux", READ | WRITE, id2)
+        .build();
+
+    assertEquals(
+        Arrays.asList(new ACL(DELETE, id1)),
+        aclProvider.getAclForPath("/foo/baz"));
```

Comment:
in case it's not clear (and this always confuses me) but hamcrest's `contains(..)` method tests not just that the item is in the collection but that the item contains the arguments in that order when iterated over